src/ObjectUtils/CoffFile.cpp

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "ObjectUtils/CoffFile.h"
+
+#include <absl/strings/str_format.h>
+#include <llvm/Demangle/Demangle.h>
+#include <llvm/Object/Binary.h>
+#include <llvm/Object/COFF.h>
+#include <llvm/Object/ObjectFile.h>
+
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/Result.h"
+#include "symbol.pb.h"
+
+namespace orbit_object_utils {
+
+namespace {
+
+using orbit_grpc_protos::ModuleSymbols;
+using orbit_grpc_protos::SymbolInfo;
+
+class CoffFileImpl : public CoffFile {
+ public:
+  CoffFileImpl(std::filesystem::path file_path,
+               llvm::object::OwningBinary<llvm::object::ObjectFile>&& owning_binary);
+  [[nodiscard]] ErrorMessageOr<orbit_grpc_protos::ModuleSymbols> LoadDebugSymbols() override;
+  [[nodiscard]] bool HasDebugSymbols() const override;
+  [[nodiscard]] std::string GetName() const override;
+  [[nodiscard]] std::filesystem::path GetFilePath() const override;
+
+ private:
+  void Initialize();
+  ErrorMessageOr<SymbolInfo> CreateSymbolInfo(const llvm::object::SymbolRef& symbol_ref);
+  const std::filesystem::path file_path_;
+  llvm::object::OwningBinary<llvm::object::ObjectFile> owning_binary_;
+  llvm::object::COFFObjectFile* object_file_;
+  bool has_symbol_table_;
+};
+
+CoffFileImpl::CoffFileImpl(std::filesystem::path file_path,
+                           llvm::object::OwningBinary<llvm::object::ObjectFile>&& owning_binary)
+    : file_path_(std::move(file_path)),
+      owning_binary_(std::move(owning_binary)),
+      has_symbol_table_(false) {
+  object_file_ = llvm::dyn_cast<llvm::object::COFFObjectFile>(owning_binary_.getBinary());
+  Initialize();
+}
+
+void CoffFileImpl::Initialize() {
```

Comment:
Are you planning on putting more stuff into this function? If not, I'd opt for putting this one line into the constructor.

Reply:
Yeah, this kind of grew out of what we have in ElfFile, but the separate method there actually has a clear meaning and it makes sense to separate it out. Here it doesn't really make sense for now. I moved the line into the constructor as you suggested.